src/System.Diagnostics.Process/src/System/Diagnostics/Process.Windows.cs

```diff
@@ -88,10 +88,26 @@ public static void LeaveDebugMode()
         /// <summary>Terminates the associated process immediately.</summary>
         public void Kill()
         {
-            using (SafeProcessHandle handle = GetProcessHandle(Interop.Advapi32.ProcessOptions.PROCESS_TERMINATE))
+            using (SafeProcessHandle handle = GetProcessHandle(Interop.Advapi32.ProcessOptions.PROCESS_TERMINATE | Interop.Advapi32.ProcessOptions.PROCESS_QUERY_LIMITED_INFORMATION, throwIfExited: false))
```

Comment:
Are there situations where the caller may have the rights necessary to use PROCESS_TERMINATE but not to use PROCESS_QUERY_LIMITED_INFORMATION?  I'm wondering if this change might be making it so that Kill stops working in cases it previously worked.

Reply:
i think only opposite is true

Reply:
TERMINATE is more impactful than QUERY_LIMITED_INFORMATION, so from that point also it makes sense if you can TERMINATE, you are likely to be able to QUERY_LIMITED_INFORMATION.
(QUERY_LIMITED_INFORMATION is also what is used to implement `HasExited`.)